lib/include/TransportDescriptor.hpp

```diff
@@ -0,0 +1,2033 @@
+/**
+ * @brief Set the SHM Transport Descriptor shared memory segment size (SHM Transport specific).
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ * @param[in] segment_size Shared memory segment size.
+ *
+ * @throw element_invalid Exception if the provided segment size is not valid.
+ */
+void set_segment_size(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id,
+        uint32_t segment_size);
+
+/**
+ * @brief Set the SHM Transport Descriptor listening port capacity (SHM Transport specific).
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ * @param[in] port_queue_capacity Listening port capacity in number of messages.
+ *
+ * @throw element_invalid Exception if the provided listening port capacity is not valid.
+ */
+void set_port_queue_capacity(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id,
+        uint32_t port_queue_capacity);
+
+/**
+ * @brief Set the SHM Transport Descriptor listening port liveliness timeout (SHM Transport specific).
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ * @param[in] healthy_check_timeout_ms Listening port liveliness timeout in milliseconds.
+ *
+ * @throw element_invalid Exception if the provided listening port liveliness timeout is not valid.
+ */
+void set_healthy_check_timeout_ms(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id,
+        uint32_t healthy_check_timeout_ms);
+
+/**
+ * @brief Set the SHM Transport Descriptor path to the debug file (SHM Transport specific).
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ * @param[in] rtps_dump_file Path to the file where RTPS messsages will be stored for debugging purposes.
+ *
+ * @throw element_invalid Exception if the provided path is not valid.
+ */
+void set_rtps_dump_file(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id,
+        const std::string& rtps_dump_file);
+
+/************************************************************************/
+/* Collection functions                                                 */
+/************************************************************************/
+
+/**
+ * @brief Append an IP address to the whitelisted network interfaces collection.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ * @param[in] ip_address IP address to be included in the whitelist collection.
+ *
+ * @throw element_invalid Exception if the provided IP address is not valid.
+ */
```

Comment:
As in `set` functions, I think that all `push` and `update` functions should also throw the `file_not_found` and `element_not_found` exceptions.

Reply:
Again, the `push` element should include the element or overwrite it if found. But it should not throw the exception. The `update` throws `element_not_found` because it is required when updating that the element can be found.